ee/middleware/upstreambasicauth/provider.go

```diff
@@ -0,0 +1,27 @@
+package upstreambasicauth
+
+import (
+	"net/http"
+
+	"github.com/sirupsen/logrus"
+)
+
+// Provider implements upstream auth provider.
+type Provider struct {
```

Comment:
This is a service struct which should have a constructor to pass the dependency (logger);
Bit weird to have it structured O(1) like this